src/charm.py

```diff
@@ -466,77 +431,16 @@ def _agent_relations(self) -> List[Relation]:
 
     @property
-    def _principal_relation(self) -> Optional[Relation]:
-        """The cos-agent relation, if the charm we're related to supports it, else juju-info."""
-        # juju relate will do "the right thing" and default to cos-agent, falling back to
-        # juju-info if no cos-agent endpoint is available on the principal.
-        # Technically, if the charm is executing, there MUST be one of these two relations
-        # (otherwise, the subordinate won't even execute). However, for the sake of juju maybe not
-        # showing us the relation until after the first few install/start/config-changed, we err on
-        # the safe side and type this as Optional.
-        principal_relations = []
-        for relation in self._agent_relations:
-            if not relation.units:
-                continue
-            if relation.name == "juju-info":
-                principal_relations.append(relation)
-                continue
-            relation_data = json.loads(
-                relation.data[next(iter(relation.units))].get("config", "{}")
-            )
-            if not relation_data:
-                continue
-            if not relation_data.get("subordinate", False):
-                principal_relations.append(relation)
-        if len(principal_relations) > 1:
-            raise MultiplePrincipalsError("Multiple Principle Applications")
-        if len(principal_relations) == 1:
-            return principal_relations[0]
-        return None
-
-    @property
-    def principal_unit(self) -> Optional[Unit]:
-        """Return the principal unit this charm is subordinated to."""
-        relation = self._principal_relation
-        if relation and relation.units:
-            # Here, we could have popped the set and put the unit back or
-            # memoized the function, but in the interest of backwards compatibility
-            # with older python versions and avoiding adding temporary state to
-            # the charm instance, we choose this somewhat unsightly option.
-            return next(iter(relation.units))
-        return None
-
-    @property
-    def principal_topology(
-        self,
-    ) -> Dict[str, str]:
-        """Return the topology of the principal unit."""
-        unit = self.principal_unit
-        if unit:
-            # Note we can't include juju_charm as that information is not available to us.
-            return {
-                "juju_model": self.model.name,
-                "juju_model_uuid": self.model.uuid,
-                "juju_application": unit.app.name,
-                "juju_unit": unit.name,
-            }
-        return {}
-
-    @property
-    def _instance_topology(self) -> Dict[str, str]:
-        return self.principal_topology
-
-    @property
-    def _principal_labels(self) -> Dict[str, str]:
-        """Return a dict with labels from the topology of the principal charm."""
+    def _instance_labels(self) -> Dict[str, str]:
```

Comment:
I'd rename it to just `labels` or `this_charm_labels`, "instance" is unclear to me

Reply:
Agreed, this was named from before. I'll look in to changing it.